### PR TITLE
fix(updater): Honor the update_critical parameter

### PR DESCRIPTION
**Pull request by jaredburkeen**
If update_critical is true, perform a silent upgrade. Otherwise, perform the normal  automatic upgrade notifications.

### CODE REVIEW

**Comment by jaredburkeen** [comment]
@loberlander This is ready for further review.

File: src/appcontroller.cpp [hunk 1]
```diff
@@ -168,7 +166,7 @@ int ApplicationController::UserRunInstallerCallback(const wchar_t* filePath)
     return ms_cbUserRunInstaller(filePath);
 }
 
-int ApplicationController::AlternateAppcastCallback(bool manual, struct Appcast& appcast)
+int ApplicationController::AlternateAppcastCallback(bool manual, struct Appcast& appcast, bool *silent)
```
Comment (loberlander) [inline src/appcontroller.cpp:169]: I think the `silent` flag belongs in `Appcast` and that would make this parameter list cleaner.

**Review by loberlander — APPROVED** [review]
Thank you for making the changes, this looks really nice.
I noted just one minor adjustment that I think still worth making.

**Review by loberlander — APPROVED** [review]
Thank you for making these final changes.

**Pull request merged**